File: LABORATORIOS/LABORATORIO2/Ejercicio4/Python/SecuenciaGenerador.py
def generar_secuencia(n):
    secuencia = []
    contador = 0

    for i in range(n):
        secuencia.append(contador)
        if contador == 0:
            contador = 1
        else:
            contador = 0

    return secuencia

n_terminos = 12
secuencia_generada = generar_secuencia(n_terminos)
print(secuencia_generada)

File: LABORATORIOS/LABORATORIO2/Ejercicio4/Java/SecuenciaGenerador.java
import java.util.ArrayList;
import java.util.List;

public class SecuenciaGenerador {
    public static List<Integer> generarSecuencia(int n) {
        List<Integer> secuencia = new ArrayList<>();
        int contador = 0;

        for (int i = 0; i < n; i++) {
            secuencia.add(contador);
            if (contador == 0) {
                contador = 1;
            } else {
                contador = 0;
            }
        }

        return secuencia;
    }

    public static void main(String[] args) {
        int nTerminos = 12;
        List<Integer> secuenciaGenerada = generarSecuencia(nTerminos);
        System.out.println(secuenciaGenerada);
    }
}